lib/url-state-machine.js

```diff
@@ -575,6 +575,14 @@ URLStateMachine.prototype["parse scheme start"] = function parseSchemeStart(c, c
 URLStateMachine.prototype["parse scheme"] = function parseScheme(c, cStr) {
   if (infra.isASCIIAlphanumeric(c) || c === p("+") || c === p("-") || c === p(".")) {
     this.buffer += cStr.toLowerCase();
+  } else if (c === p(":") && this.buffer.length === 1 &&
+             [...this.buffer].every(ch => infra.isASCIIAlpha(ch.codePointAt(0))) &&
+             this.input[this.pointer + 1] === p("\\")) {
+    this.url.scheme = "file";
+    this.url.host = "";
+    // Keep the original case from input for Windows drive letter
+    this.buffer = `${String.fromCodePoint(this.input[this.pointer - 1])}:`;
```

Comment:
This step still does not appear in the spec. If you were to translate the spec it would be

```js
this.buffer = this.buffer + ":";
```

If this doesn't work, then the spec is wrong.

Reply:
thanks, I updated spec (https://github.com/whatwg/url/pull/874/commits/b98fc9b7539717fd5cabf9eddc474351a6af0baf)

Reply:
OK, well, now the code here doesn't match the updated spec, and also the tests don't pass. So let me know when you've achieved the goal of aligned code and spec, plus passing tests, at the same time.